src/aalwines/model/Router.cpp

```diff
@@ -44,6 +44,12 @@ namespace aalwines
         _names.emplace_back(name);
     }
 
+    void Router::change_name(const std::string& name)
```

Comment:
I am not sure this is a good idea.
I think I would prefer
```
_names.clear();
_names.emplace_back(name);
```
to avoid a few corner-cases.

Maybe also add
```
assert(_names.size() == 1);
```

Reply:
Done.

I think it makes more sense to have just one std::string _name instead of an std::vector of names. As far as I can see, that is all it needs. But that discussion might be for a future 'Refactoring' pull-request.

Reply:
The short answer is that the juniper configurations allow for multiple names for a single router (e.g. IPv4, IPv6, and different labels). You can use any of them in the query.
For convenience we only use the last during I/O of the tool.